hs_core/management/commands/solr_recover.py

```diff
@@ -0,0 +1,125 @@
+    baseinstance = BaseResource.objects.get(pk=instance.pk)
+    basesender = BaseResource
+    using_backends = connection_router.for_write(instance=baseinstance)
+    for using in using_backends:
+        # if object is public/discoverable or becoming public/discoverable, index it
+        if instance.raccess.public or instance.raccess.discoverable:
+            try:
+                index = connections[using].get_unified_index().get_index(basesender)
+                index.update_object(baseinstance, using=using)
+            except NotHandled:
+                logger.exception(
+                    "Failure: changes to %s with short_id %s not added to Solr Index.",
+                    str(type(instance)), baseinstance.short_id)
+
+        # if object is private or becoming private, delete from index
+        else:
+            try:
+                index = connections[using].get_unified_index().get_index(basesender)
+                index.remove_object(baseinstance, using=using)
+            except NotHandled:
+                logger.exception("Failure: delete of %s with short_id %s failed.",
+                                 str(type(instance)), baseinstance.short_id)
+
+
+class Command(BaseCommand):
+    help = "Print size information."
+
+    def add_arguments(self, parser):
+
+        # a list of resource id's: none does nothing.
+        parser.add_argument('resource_ids', nargs='*', type=str)
+
+        # Named (optional) arguments
+        parser.add_argument(
+            '--log',
+            action='store_true',  # True for presence, False for absence
+            dest='log',           # value is options['log']
+            help='log errors to system log',
+        )
+
+        parser.add_argument(
+            '--type',
+            dest='type',
+            help='limit to resources of a particular type'
+        )
+
+        parser.add_argument(
+            '--storage',
+            dest='storage',
+            help='limit to specific storage medium (local, user, federated)'
+        )
+
+        parser.add_argument(
+            '--access',
+            dest='access',
+            help='limit to specific access class (public, discoverable, private)'
+        )
+
+        parser.add_argument(
+            '--has_subfolders',
+            action='store_true',  # True for presence, False for absence
+            dest='has_subfolders',  # value is options['has_subfolders']
+            help='limit to resources with subfolders',
+        )
+
+    def repair_filtered_solr(self, resource, options):
+        if (options['type'] is None or resource.resource_type == options['type']) and \
+           (options['storage'] is None or resource.storage_type == options['storage']) and \
+           (options['access'] != 'public' or resource.raccess.public) and \
+           (options['access'] != 'discoverable' or resource.raccess.discoverable) and \
+           (options['access'] != 'private' or not resource.raccess.discoverable) and \
```

Comment:
you've got a bug with private access checking discoverable... We could probably just scrap the access options for now?